server/src/main/java/org/elasticsearch/search/aggregations/InternalOrder.java

```diff
@@ -153,8 +153,8 @@ public AggregationPath path() {
 
             @Override
             public int compare(Bucket b1, Bucket b2) {
-                double v1 = path.resolveValue(b1);
-                double v2 = path.resolveValue(b2);
+                double v1 = path.resolveValue(((InternalAggregations) b1.getAggregations()));
+                double v2 = path.resolveValue(((InternalAggregations) b2.getAggregations()));
```

Comment:
This feels a little sneaky, but nowhere near as sneaky as the old `instanceof`s. And this stuff seems to only be used internally so I think we're safe here.